actors/builtin/multisig/multisig_test.go

```diff
@@ -311,6 +310,134 @@ func TestApprove(t *testing.T) {
 	})
 }
 
+func TestCancel(t *testing.T) {
+	actor := msActorHarness{multisig.MultiSigActor{}, t}
+
+	receiver := newIDAddr(t, 100)
+	anne := newIDAddr(t, 101)
+	bob := newIDAddr(t, 102)
+	chuck := newIDAddr(t, 103)
+
+	richard := newIDAddr(t, 104)
+
+	const noUnlockDuration = int64(0)
+	const numApprovals = int64(2)
+	const txnID = int64(0)
+	var sendValue = abi.NewTokenAmount(10)
+	var nilParams = abi.MethodParams{}
+	var signers = []addr.Address{anne, bob}
+
+	builder := mock.NewBuilder(context.Background(), t, receiver).WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID)
+
+	t.Run("simple propose and cancel", func(t *testing.T) {
+		rt := builder.Build()
+
+		actor.constructAndVerify(rt, numApprovals, noUnlockDuration, signers...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		serParams := serde.MustSerialize(&multisig.ChangeNumApprovalsThresholdParams{NewThreshold: 1000})
+		actor.propose(rt, chuck, sendValue, builtin.Method_MultiSigActor_ChangeNumApprovalsThreshold, serParams)
```

Comment:
This is testing changing the number of approvals, which is orthogonal to cancellation. Remove this from this test, put it somewhere else. Such a test should verify that the new approval threshold is respected, by approving a transaction.

What happens if the threshold is reduced to a number that would cause some transaction/s to meet it? Sounds like more undefined behaviour.

Reply:
Oooooh, this is really confusing because the proposed method is also on the multisig. That makes it really hard to follow. Please just use a SomeMethod=42 constant, and an arbitrary parameter object just for testing, nothing to do with multisig.

Reply:
In a [previous PR](https://github.com/filecoin-project/specs-actors/pull/58#discussion_r371912000) the suggestion was made to use non-zero methods and parameters here. Since the Multisig actor is the only actor with params I just grabbed them. These same parameters are also used on [lines above](https://github.com/filecoin-project/specs-actors/pull/63/files#diff-d5777d603b48d25386edace2468db7cfR202) as well. I'm going to change this, and the line above back to the zero method and nilParams since this makes the test hard to follow.